src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -1173,6 +1181,76 @@ std::pair<std::shared_ptr<Pane>, std::shared_ptr<Pane>> Pane::_Split(SplitState
     return { _firstChild, _secondChild };
 }
 
+// Method Description:
+// - Recursively attempt to "zoom" the given pane. When the pane is zoomed, it
+//   won't be displayed as part of the tab tree, instead it'll take up the full
+//   content of the tab. When we find the given pane, we'll need to remove it
+//   from the UI tree, so that the caller can re-add it. We'll also set some
+//   internal state, so the pane can display all of its borders.
+// Arguments:
+// - zoomedPane: This is the pane which we're attempting to zoom on.
+// Return Value:
+// - <none>
+void Pane::Zoom(std::shared_ptr<Pane> zoomedPane)
+{
+    if (_IsLeaf())
+    {
+        _zoomed = (zoomedPane == shared_from_this());
+        _UpdateBorders();
+    }
+    else
+    {
+        if (zoomedPane == _firstChild || zoomedPane == _secondChild)
+        {
+            // When we're zooming the pane, we'll need to remove it from our UI
+            // tree. Easy way: just remove both children. We'll re-attach both
+            // when we un-zoom.
+            _root.Children().Clear();
+        }
+
+        // Always recurse into both children. If the (un)zoomed pane was one of
+        // our direct children, we'll still want to update it's borders.
+        _firstChild->Zoom(zoomedPane);
+        _secondChild->Zoom(zoomedPane);
+    }
+}
+
+// Method Description:
+// - Recursively attempt to "un-zoom" the given pane. This does the opposite of
+//   Pane::Zoom. When we find the given pane, we should return the pane to our
+//   UI tree. We'll also clear the internal state, so the pane can display its
+//   borders correctly.
+// - The caller should make sure to have removed the zoomed pane from the UI
+//   tree _before_ calling this.
+// Arguments:
+// - zoomedPane: This is the pane which we're attempting to un-zoom.
+// Return Value:
+// - <none>
+void Pane::UnZoom(std::shared_ptr<Pane> zoomedPane)

```

Comment:
Shoulda been `Enzoom` and `Dezoom` like the good ol' `Embolden` and `Debolden` that I think @j4james took out. :P

Reply:
If it's any consolation, Embolden and Debolden still live on in the [comments](https://github.com/microsoft/terminal/blob/1f8264d86b39a83f8cae26be0f7a006d7ebf16a9/src/renderer/vt/VtSequences.cpp#L339).

And FWIW, I thought your suggestion of Maximize/Restore for the method names was more meaningful. My first impression from Zoom/UnZoom was that this had something to do with font scaling.